CHANGELOG.md

```diff
@@ -4,6 +4,72 @@ All notable changes to Router will be documented in this file.
+* `compute_job.execution`
+  - `job.age`: `P1`-`P8`
+  - `job.type`: (`query_parsing`|`query_planning`|`introspection`)
+
+Jobs are executed highest priority (`P8`) first. Jobs that are low priority (`P1`) age over time, eventually executing 
+at highest priority. The age of a job is can be used to diagnose if a job was waiting in the queue due to other higher 
+priority jobs also in the queue.
+
+By [@bryncooke](https://github.com/bryncooke) in https://github.com/apollographql/router/pull/7236
+
+### Add compute job pool metrics ([PR #7184](https://github.com/apollographql/router/pull/7184))
+
+The compute job pool is used within the router for compute intensive jobs that should not block the Tokio worker threads.
+When this pool becomes saturated it is difficult for users to see why so that they can take action.
+This change adds new metrics to help users understand how long jobs are waiting to be processed.  
+
+New metrics:
+- `apollo.router.compute_jobs.queue_is_full` - A counter of requests rejected because the queue was full.
+- `apollo.router.compute_jobs.duration` - A histogram of time spent in the compute pipeline by the job, including the queue and query planning.
+  - `job.type`: (`query_planning`, `query_parsing`, `introspection`)
+  - `job.outcome`: (`executed_ok`, `executed_error`, `channel_error`, `rejected_queue_full`, `abandoned`)
+- `apollo.router.compute_jobs.queue.wait.duration` - A histogram of time spent in the compute queue by the job.
+  - `job.type`: (`query_planning`, `query_parsing`, `introspection`)
+- `apollo.router.compute_jobs.execution.duration` - A histogram of time spent to execute job (excludes time spent in the queue).
+  - `job.type`: (`query_planning`, `query_parsing`, `introspection`)
+- `apollo.router.compute_jobs.active_jobs` - A gauge of the number of compute jobs being processed in parallel.
+  - `job.type`: (`query_planning`, `query_parsing`, `introspection`)
+
+By [@carodewig](https://github.com/carodewig) in https://github.com/apollographql/router/pull/7184
+
+## 🐛 Fixes
+
+### Poll pending compute jobs hang request ([PR #7273](https://github.com/apollographql/router/pull/7273))
+
+Compute jobs in the router are used to execute CPU intensive work outside of the main I/O worker threads, including GraphQL parsing, query planning, and introspection.
+
+We previously checked if the compute job queue was full and returned `Poll::Pending` in the tower services.
+However, this will cause requests to hang until timeout.
+
+This PR shifts the logic into `call` and will immediately return a `SERVICE_UNAVAILABLE` response to the user.
+
+By [@BrynCooke](https://github.com/BrynCooke) in https://github.com/apollographql/router/pull/7273
+
+### Increase compute job pool queue size ([PR #7205](https://github.com/apollographql/router/pull/7205))
+
+Compute jobs in the router are used to execute CPU intensive work outside of the main I/O worker threads, including GraphQL parsing, query planning, and introspection.
```

Comment:
Standardise on "compute job pool" + introduce the queue that the next line refers to
```suggestion
The compute job pool in the router is used to execute CPU intensive work outside of the main I/O worker threads, including GraphQL parsing, query planning, and introspection. When the pool is busy, jobs enter a queue.
```